libvmi/os/windows/core.c

```diff
@@ -917,6 +917,7 @@ init_from_rekall_profile_real(vmi_instance_t vmi, reg_t kpcr_register_to_use)
         dbprint(VMI_DEBUG_MISC, "**KernBase PA=0x%"PRIx64"\n", windows->ntoskrnl);
 
         if (windows->ntoskrnl && kdvb && kernbase_offset) {
+            /* This could fail because the address mode is unknown. */
             if ( VMI_FAILURE == vmi_read_addr_pa(vmi, windows->ntoskrnl + kdvb + kernbase_offset, &windows->ntoskrnl_va) )
                 goto done;
```

Comment:
Although I didn't change this block, the logic seems wrong. If a read fails, we bail. Right after that, we try the read again.

In the case of a file, the first read will fail because we don't have the addr mode.